<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>985</width>
    <height>746</height>
   </rect>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>-1</number>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>985</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_New"/>
    <addaction name="action_Open"/>
    <addaction name="action_Save"/>
    <addaction name="action_Close"/>
    <addaction name="separator"/>
    <addaction name="actionUpPastebin"/>
    <addaction name="separator"/>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menu_Edit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="action_Copy"/>
    <addaction name="action_Cut"/>
    <addaction name="action_Paste"/>
    <addaction name="separator"/>
    <addaction name="action_Find"/>
   </widget>
   <widget class="QMenu" name="menu_Settings">
    <property name="title">
     <string>&amp;Settings</string>
    </property>
    <addaction name="action_Configure_the_IDE"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <widget class="QMenu" name="menuCommunity">
     <property name="title">
      <string>Community</string>
     </property>
     <property name="icon">
      <iconset resource="../resources/images.qrc">
       <normaloff>:/images/arduide.png</normaloff>:/images/arduide.png</iconset>
     </property>
     <addaction name="actionCommunityArduinoCC"/>
     <addaction name="actionCommunityArduinoForums"/>
     <addaction name="separator"/>
    </widget>
    <addaction name="action_Contextual_help"/>
    <addaction name="separator"/>
    <addaction name="menuCommunity"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="action_Utilities"/>
   </widget>
   <widget class="QMenu" name="menu_Libraries">
    <property name="title">
     <string>Libraries</string>
    </property>
    <addaction name="action_Lib_Refresh"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Edit"/>
   <addaction name="menu_Libraries"/>
   <addaction name="menu_View"/>
   <addaction name="menu_Settings"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>Edition</string>
   </property>
   <attribute name="toolBarArea">
    <enum>BottomToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_New"/>
   <addaction name="action_Open"/>
   <addaction name="action_Save"/>
   <addaction name="action_Close"/>
   <addaction name="separator"/>
   <addaction name="action_Copy"/>
   <addaction name="action_Cut"/>
   <addaction name="action_Paste"/>
   <addaction name="separator"/>
   <addaction name="action_Prev"/>
   <addaction name="action_Next"/>
  </widget>
  <widget class="QToolBar" name="deviceToolBar">
   <property name="windowTitle">
    <string>Hardware</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>BottomToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QToolBar" name="toolBar_2">
   <property name="windowTitle">
    <string>Build tools</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>BottomToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Build"/>
   <addaction name="action_Upload"/>
   <addaction name="action_Utilities"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="features">
    <set>QDockWidget::DockWidgetFeatureMask</set>
   </property>
   <property name="windowTitle">
    <string>Utility dock</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <property name="sizeConstraint">
      <enum>QLayout::SetMinimumSize</enum>
     </property>
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <widget class="QTabWidget" name="utilityTabWidget">
       <property name="tabPosition">
        <enum>QTabWidget::East</enum>
       </property>
       <property name="tabShape">
        <enum>QTabWidget::Rounded</enum>
       </property>
       <widget class="QWidget" name="tab">
        <attribute name="icon">
         <iconset resource="../resources/images.qrc">
          <normaloff>:/images/16x16/terminal.png</normaloff>:/images/16x16/terminal.png</iconset>
        </attribute>
        <attribute name="title">
         <string>Output</string>
        </attribute>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="sizeConstraint">
          <enum>QLayout::SetMinimumSize</enum>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="OutputView" name="outputView">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(0, 0, 0);</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockFindReplace">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="maximumSize">
    <size>
     <width>524287</width>
     <height>90</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Find/Replace</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="sizeConstraint">
      <enum>QLayout::SetMinimumSize</enum>
     </property>
     <item>
      <layout class="QGridLayout" name="gridLayout">
       <property name="sizeConstraint">
        <enum>QLayout::SetMinimumSize</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="labelFind">
         <property name="text">
          <string>Find</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="labelReplace">
         <property name="text">
          <string>Replace</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="lineFind">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="lineReplace">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="pushFind">
         <property name="text">
          <string>Find</string>
         </property>
         <property name="default">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="pushReplace">
         <property name="text">
          <string>Replace</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QPushButton" name="pushReplaceAll">
         <property name="text">
          <string>Replace All</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QCheckBox" name="checkRegExp">
         <property name="text">
          <string>Reg. Exp.</string>
         </property>
        </widget>
       </item>
       <item row="2" column="4">
        <widget class="QCheckBox" name="checkWordOnly">
         <property name="text">
          <string>Word only</string>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QCheckBox" name="checkCaseSensitive">
         <property name="text">
          <string>Case sensitive</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="action_Quit">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/application-exit.png</normaloff>:/images/16x16/application-exit.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_New">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/document-new.png</normaloff>:/images/16x16/document-new.png</iconset>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_Copy">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/edit-copy.png</normaloff>:/images/16x16/edit-copy.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="action_Paste">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/edit-paste.png</normaloff>:/images/16x16/edit-paste.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="action_Cut">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/edit-cut.png</normaloff>:/images/16x16/edit-cut.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Cut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/document-save.png</normaloff>:/images/16x16/document-save.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_Open">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/document-open.png</normaloff>:/images/16x16/document-open.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_Close">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/document-close.png</normaloff>:/images/16x16/document-close.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="action_Build">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/run-build.png</normaloff>:/images/16x16/run-build.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Build (verify)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="action_Utilities">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/applications-development.png</normaloff>:/images/16x16/applications-development.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Utilities</string>
   </property>
   <property name="toolTip">
    <string>Utilities</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="action_Upload">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/run-build-install.png</normaloff>:/images/16x16/run-build-install.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Upload</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
  <action name="actionGo_to_the_next_tab">
   <property name="text">
    <string>Go to the next tab</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+PgDown</string>
   </property>
  </action>
  <action name="actionGo_to_the_previous_tab">
   <property name="text">
    <string>Go to the previous tab</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+PgUp</string>
   </property>
  </action>
  <action name="action_Configure_the_IDE">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/configure.png</normaloff>:/images/16x16/configure.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Configure the IDE</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/help-about.png</normaloff>:/images/16x16/help-about.png</iconset>
   </property>
   <property name="text">
    <string>&amp;About %0</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/help-about.png</normaloff>:/images/16x16/help-about.png</iconset>
   </property>
   <property name="text">
    <string>About &amp;Qt</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/edit-undo.png</normaloff>:/images/16x16/edit-undo.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/edit-redo.png</normaloff>:/images/16x16/edit-redo.png</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="action_Prev">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/go-previous.png</normaloff>:/images/16x16/go-previous.png</iconset>
   </property>
   <property name="text">
    <string>Previous</string>
   </property>
   <property name="toolTip">
    <string>Previous</string>
   </property>
  </action>
  <action name="action_Next">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/go-next.png</normaloff>:/images/16x16/go-next.png</iconset>
   </property>
   <property name="text">
    <string>Next</string>
   </property>
   <property name="toolTip">
    <string>Next</string>
   </property>
  </action>
  <action name="action_Contextual_help">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/help-contextual.png</normaloff>:/images/16x16/help-contextual.png</iconset>
   </property>
   <property name="text">
    <string>Contextual help</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionCommunityArduinoCC">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/arduide.png</normaloff>:/images/arduide.png</iconset>
   </property>
   <property name="text">
    <string>The official arduino website</string>
   </property>
  </action>
  <action name="actionCommunityArduinoForums">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/arduide.png</normaloff>:/images/arduide.png</iconset>
   </property>
   <property name="text">
    <string>The arduino forums</string>
   </property>
  </action>
  <action name="actionUpPastebin">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/edit-paste.png</normaloff>:/images/16x16/edit-paste.png</iconset>
   </property>
   <property name="text">
    <string>Upload to pastebin</string>
   </property>
  </action>
  <action name="action_Find">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/edit-find-replace.png</normaloff>:/images/16x16/edit-find-replace.png</iconset>
   </property>
   <property name="text">
    <string>Find/Replace</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="action_Lib_Refresh">
   <property name="icon">
    <iconset resource="../resources/images.qrc">
     <normaloff>:/images/16x16/view-refresh.png</normaloff>:/images/16x16/view-refresh.png</iconset>
   </property>
   <property name="text">
    <string>Refresh</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>OutputView</class>
   <extends>QTextBrowser</extends>
   <header>gui/OutputView.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_Quit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
